Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -711,6 +723,10 @@ public class DependencyResolver<
     ) -> AnySequence<AssignmentSet> {
         var allConstraints = allConstraints
 
+        // Ask all of these containers up-front to allow for async cloning, if supported
+        // by provider.
+        provider.mayNeed(containers: constraints.map{$0.identifier})
```

Comment:
Random performance nit: this constructs a temporary array just to pass to the function, which is why it is more efficient to just ask one by one (previous comment).